File: 10814.py
N = int(input())
list = []
for i in range(N):
    age, name = input().split()
    list.append([age, name, i])
# for i in range(N):
#     for j in range(N):
#         if list[j][0] > list[i][0]:
#             temp = list[j][0]
#             list[j][0] = list[i][0]
#             list[i][0] = temp
list.sort(key=lambda x: (int(x[0]), int(x[2])))
for i in range(N):
    print(list[i][0], list[i][1])

File: BOJ/10814.cpp
#include <iostream>
#include <vector>
#include <string>
#include <utility>

using namespace std;

int main()
{
    typedef pair<int, string> pp;
    typedef pair<int, pp> p;

    int N;
    vector<p> v;

    cin >> N;
    int age;
    string name;
    for (int i = 0; i < N; i++)
    {
        cin >> age >> name;
        pp mypp = make_pair(age, name);
        p myp = make_pair(i, mypp);
        v.push_back(myp);
    }

    for (int i = 0; i < v.size(); i++)
    {
        for (int j = i + 1; j < v.size(); j++)
        {
            if (v[i].second.first > v[j].second.first)
            {
                pp temp;
                temp = v[i];
                v[i] = v[j];
                v[j] = temp;
            }
        }
    }
    for (int i = 0; i < v.size(); i++)
    {
        for (int j = i + 1; j < v.size(); j++)
        {

            if (v[i].second.first == v[j].second.first)
            {
                if (v[i].first > v[j].first)
                {
                    p temp;
                    temp = v[i];
                    v[i] = v[j];
                    v[j] = temp;
                }
            }
        }
    }

    for (int i = 0; i < v.size(); i++)
    {
        cout << v[i].second.first << " " << v[i].second.second << '\n';
    }
}